bundler/lib/bundler/plugin.rb

```diff
@@ -342,7 +342,30 @@ def load_plugin(name)
       # done to avoid conflicts
       path = index.plugin_path(name)
 
-      Gem.add_to_load_path(*index.load_paths(name))
+      paths = index.load_paths(name)
+
+      all_paths_valid = true
+      paths.each do |p|
+        next if File.directory?(p)
+
+        message = <<~MESSAGE
+          Failed loading plugin #{name}:
+
+          Plugin path #{p} does not exist. This can happen if the plugin was \
+          installed with a different version of Ruby that has since been uninstalled.
+
+          If you would like to reinstall the plugin, run:
+
+          bundler plugin uninstall #{name} && bundler plugin install #{name}
+
+          Continuing without installing plugin #{name}.
+        MESSAGE
+        Bundler.ui.warn message
+        all_paths_valid = false
+      end
+      return unless all_paths_valid
+
+      Gem.add_to_load_path(*paths)
```

Comment:
If there are two non existing paths in the same plugin, we'll end up printing this big error block twice. I think it will read slightly better to find all invalid paths first, and then print an error including all of them. Like

```ruby
      paths = index.load_paths(name)
      invalid_paths = paths.reject {|p| File.directory?(p) }

      if invalid_paths.any?
        Bundler.ui.warn <<~MESSAGE
          The following plugin paths don't exist: #{invalid_paths.join(", ")}.

          This can happen if the plugin was installed with a different version of Ruby that has since been uninstalled.

          If you would like to reinstall the plugin, run:
          
          bundler plugin uninstall #{name} && bundler plugin install #{name}

          Continuing without installing plugin #{name}.
        MESSAGE

        return
      end

      Gem.add_to_load_path(*paths)
```

Reply:
Ya, I agree that makes sense. I updated code/test. This is what it looks like now after running `rake setup`:

![Screenshot 2024-09-13 at 12 48 29 PM](https://github.com/user-attachments/assets/22a22e7f-d9b7-41e6-a3e2-aa272bdc1f42)